vfs/directory.go

```diff
@@ -363,6 +367,19 @@ func bulkUpdateDocsPath(c Context, oldpath, newpath string) error {
 	return err
 }
 
+// TrashDir is used to delete a directory given its document
+func TrashDir(c Context, olddoc *DirDoc) (newdoc *DirDoc, err error) {
+	trashFolderID := TrashFolderID
+	err = tryOrUseSuffix(olddoc.Name, "%scozy__%s", func(name string) error {
+		newdoc, err = ModifyDirMetadata(c, olddoc, &DocPatch{
```

Comment:
Why do you assign `newdoc` inside the anonymous func but return `err` to assign it from `TrashDir`?